app/javascript/dashboard/modules/search/components/RecentSearches.vue

```diff
@@ -0,0 +1,123 @@
+<script setup>
+import { ref, computed, onMounted } from 'vue';
+import { LocalStorage } from 'shared/helpers/localStorage';
+import { LOCAL_STORAGE_KEYS } from 'dashboard/constants/localStorage';
+
+import Icon from 'dashboard/components-next/icon/Icon.vue';
+import Button from 'dashboard/components-next/button/Button.vue';
+
+const emit = defineEmits(['selectSearch', 'clearAll']);
+
+const MAX_RECENT_SEARCHES = 3;
+const recentSearches = ref([]);
+
+const loadRecentSearches = () => {
+  const stored = LocalStorage.get(LOCAL_STORAGE_KEYS.RECENT_SEARCHES) || [];
+  recentSearches.value = Array.isArray(stored) ? stored : [];
+};
+
+const saveRecentSearches = () => {
+  LocalStorage.set(LOCAL_STORAGE_KEYS.RECENT_SEARCHES, recentSearches.value);
+};
+
+const addRecentSearch = query => {
+  if (!query || query.trim().length < 2) return;
+
+  const trimmedQuery = query.trim();
+
+  const existingIndex = recentSearches.value.findIndex(
+    search => search.toLowerCase() === trimmedQuery.toLowerCase()
+  );
+
+  if (existingIndex !== -1) {
+    recentSearches.value.splice(existingIndex, 1);
+  }
+
+  recentSearches.value.unshift(trimmedQuery);
+
+  if (recentSearches.value.length > MAX_RECENT_SEARCHES) {
+    recentSearches.value = recentSearches.value.slice(0, MAX_RECENT_SEARCHES);
+  }
+
+  saveRecentSearches();
+};
+
+const clearRecentSearches = () => {
+  recentSearches.value = [];
+  LocalStorage.remove(LOCAL_STORAGE_KEYS.RECENT_SEARCHES);
+};
+
+const hasRecentSearches = computed(() => recentSearches.value?.length > 0);
+
+const onSelectSearch = query => {
+  emit('selectSearch', query);
+};
+
+const onClearAll = () => {
+  clearRecentSearches();
+  emit('clearAll');
+};
+
+defineExpose({
+  addRecentSearch,
+  loadRecentSearches,
+});
+
+onMounted(() => {
+  loadRecentSearches();
+});
+</script>
+
+<template>
+  <div v-if="hasRecentSearches" class="px-4 pb-4 w-full pt-2">
+    <div class="flex items-center justify-between mb-4">
+      <div class="flex items-center gap-2.5">
+        <Icon icon="i-lucide-rotate-ccw" class="text-n-slate-10 size-4" />
+        <h3 class="text-base font-medium text-n-slate-10">
+          {{ $t('SEARCH.RECENT_SEARCHES') }}
+        </h3>
+      </div>
+      <Button
+        v-if="hasRecentSearches"
```

Comment:
Redundant condition since it is already inside a block that already requires hasRecentSearches to be true

Reply:
Done https://github.com/chatwoot/chatwoot/pull/12904/commits/68aa21fb9bc46e15c62dc913e3c603d3e11d2d6b